### PR TITLE
add encode ID API endpoint

### LINKED ISSUE #17503: Encode ID API endpoint missing
**Describe the bug**
Not really a bug, but a feature that is missing.
If I want to interact with the galaxy API and for example delete users, like the [WALL·E script](https://github.com/usegalaxy-eu/WallE) is doing, you need to be able to encode galaxy IDs.
You could do this with the `secret_decoder_ring.py`, but you would need to open subprocess and source Galaxy's virtualenv and pass Galaxy's config file with the id_secret.
This seems to be a bit hacky to me and there might be scenarios where WALL·E runs on nodes, that can not access this secret and/or Galaxy's virtualenv. (Even though I would need to change a few things to reduce dependencis for sure.)

EDIT: I forgot that it actually sources the virtualenv currently, because I can avoid installing packages twice then, so yeah, maybe I can also do it the hacky way and we can close this issue.
(if not someone else finds this endpoint useful and wants to implement it)

I understand that this is not an important feature to implement, but maybe it could serve as a good first issue.

**Expected behavior**
An API endpoint like
`/api/configuration/encode/{decoded_id}`

### PR DESCRIPTION
**Pull request by mira-miracoli**
fixes #17503 

## How to test the changes?
(Select all options that apply)

- [ ] Instructions for manual testing are as follows:
  1. request the newly added api

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/configuration.py [hunk 1]
```diff
@@ -39,6 +39,11 @@
     description="Encoded id to be decoded",
 )
 
+DecodedIdPathParam = Path(
+    ...,
+    title="Decoded id",
+    description="Decoded id to be Encoded",
```
Comment (mira-miracoli) [inline lib/galaxy/webapps/galaxy/api/configuration.py]:
```suggestion
    description="Decoded id to be encoded",
```

**Comment by mira-miracoli** [comment]
I tried running `make build-api-schema` but get an error:
~~~
  File "/home/mk/repos/galaxy/lib/galaxy/webapps/galaxy/fast_app.py", line 104
    if x_frame_options := gx_app.config.x_frame_options:
                        ^
SyntaxError: invalid syntax
~~~
It seems that the Makefile uses python 3.7.12, which doesn't know about the walrus operator yet

File: _schema.yaml
Comment (hexylena) [inline _schema.yaml:1]: not sure if these should be included?

File: lib/galaxy/managers/configuration.py [hunk 2]
```diff
@@ -56,6 +57,14 @@ def decode_id(
         decoded_id = self._app.security.decode_id(encoded_id)
         return {"decoded_id": decoded_id}
 
+    def encode_id(
+        self,
+        decoded_id: int,
+    ) -> Dict[str, str]:
+        # Handle the special case for library folders
```
Comment (hexylena) [inline lib/galaxy/managers/configuration.py]:
```suggestion
```

File: _schema.yaml
Comment (mira-miracoli) [reply _schema.yaml:1]: The GH action told me so 😳 
("run `make build-api-schema` and commit results")

Comment (davelopez) [reply _schema.yaml:1]: you should run `make update-client-api-schema`, that will get rid of these temporal files and generate the `schema.ts` for the client

File: lib/galaxy/managers/configuration.py [hunk 3]
```diff
@@ -56,6 +57,13 @@ def decode_id(
         decoded_id = self._app.security.decode_id(encoded_id)
         return {"decoded_id": decoded_id}
 
+    def encode_id(
+        self,
+        decoded_id: int,
+    ) -> Dict[str, str]:
+        encoded_id = unicodify(self._app.security.encode_id(decoded_id))
```
Comment (davelopez) [inline lib/galaxy/managers/configuration.py]: The result of `self._app.security.encode_id` will be already "unicodified" so you can safely remove this call and the import.

https://github.com/galaxyproject/galaxy/blob/566731bdb0f80d45aa3d669b2a6fd0024af8fb78/lib/galaxy/security/idencoding.py#L49

```suggestion
        encoded_id = self._app.security.encode_id(decoded_id)
```

**Review by davelopez — APPROVED** [review]
Thank you @mira-miracoli looks good!

**Pull request merged**